src/main/java/org/folio/circulation/resources/CheckInByBarcodeResource.java

```diff
@@ -83,7 +83,8 @@ private void checkin(RoutingContext routingContext) {
       .thenComposeAsync(r -> r.after(
         records -> processAdapter.createOverdueFineIfNecessary(records, context)))
       .thenApply(r -> r.next(requestScheduledNoticeService::rescheduleRequestNotices))
-      .thenApply(CheckInByBarcodeResponse::from)
-      .thenAccept(r -> r.writeTo(routingContext.response()));
+      .thenApply(r -> r.map(CheckInByBarcodeResponse::fromRecords))
+      .thenApply(r -> r.map(CheckInByBarcodeResponse::toHttpResponse))
+      .thenAccept(r -> r.applySideEffect(context::write, context::write));
```

Comment:
Does it make sense to move this method somewhere and name it to describe that it is used to write the result?

Reply:
> Does it make sense to move this method somewhere and name it to describe that it is used to write the result?

I'm not sure I follow. Which method are you referring to moving or renaming?

Reply:
the `r.applySideEffect(context::write, context::write)` method.

Reply:
@bohdan-suprun I've extracted an extra method on the context. What do you think?

I originally named it `writeResponseAsSideEffect`. I felt that was wordy and didn't really add much. Thinking about it now, I could rename it to something like `writeResultToHttpResponse`.

What name would make the most sense to you?

Reply:
The `writeResultToHttpResponse` sounds clear to me.

Reply:
Ok, I'll rename it to that. Thanks